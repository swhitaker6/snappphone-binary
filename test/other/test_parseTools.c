#include <limits.h>
#include <stdio.h>
#include <inttypes.h>

void test_makeGetValue(int32_t* ptr);
int  test_receiveI64ParamAsI53(int64_t arg1, int64_t arg2);
int  test_receiveI64ParamAsDouble(int64_t arg1, int64_t arg2);

#define MAX_SAFE_INTEGER (1ll << 53)
#define MIN_SAFE_INTEGER (-MAX_SAFE_INTEGER)

int main() {
  int rtn;
  printf("MAX_SAFE_INTEGER: %lld\n", MAX_SAFE_INTEGER);
  printf("MIN_SAFE_INTEGER: %lld\n", MIN_SAFE_INTEGER);

  int32_t num = -0x12345678;
  test_makeGetValue(&num);

  rtn = test_receiveI64ParamAsI53(42, -42);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsI53(MAX_SAFE_INTEGER, MIN_SAFE_INTEGER);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsI53(MAX_SAFE_INTEGER + 1, 0);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsI53(MIN_SAFE_INTEGER - 1, 0);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsDouble(42, -42);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsDouble(MAX_SAFE_INTEGER, MIN_SAFE_INTEGER);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsDouble(MAX_SAFE_INTEGER + 1, 0);
  printf("rtn = %d\n", rtn);

  rtn = test_receiveI64ParamAsDouble(MIN_SAFE_INTEGER - 1, 0);
  printf("rtn = %d\n", rtn);

  printf("done\n");
  return 0;
}
