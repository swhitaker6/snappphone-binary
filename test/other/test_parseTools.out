MAX_SAFE_INTEGER: 9007199254740992
MIN_SAFE_INTEGER: -9007199254740992

test_makeGetValue:
i32: -12345678
u32: edcba988
u32 legacy: edcba988
i16: -5678
u16: a988
u16 legacy: a988
i8: -78
u8: 88
u8 legacy: 88
ptr: edcba988
ptr: edcba988

test_receiveI64ParamAsI53:
arg1: 42
arg2: -42
rtn = 0

test_receiveI64ParamAsI53:
arg1: 9007199254740992
arg2: -9007199254740992
rtn = 0

test_receiveI64ParamAsI53:
rtn = -1

test_receiveI64ParamAsI53:
rtn = -1

test_receiveI64ParamAsDouble:
arg1: 42
arg2: -42
rtn = 0

test_receiveI64ParamAsDouble:
arg1: 9007199254740992
arg2: -9007199254740992
rtn = 0

test_receiveI64ParamAsDouble:
arg1: 9007199254740992
arg2: 0
rtn = 0

test_receiveI64ParamAsDouble:
arg1: -9007199254740992
arg2: 0
rtn = 0
done
